### PR TITLE
Pull CI from a remote gist (need to replace with shared-step)

### PR DESCRIPTION
**Pull request by stjohnjohnson**
Source is here: https://gist.github.com/stjohnjohnson/3d2388b2a7ba658cdcdaffa8cd874e50

This should make it easier to apply this everywhere.  Also uses ssh instead of https token for git push.

### CODE REVIEW

File: screwdriver.yaml [hunk 1]
```diff
@@ -4,13 +4,17 @@ jobs:
...
+            - publish-npm: ./ci/publish.sh
+            - publish-docker: ./ci/docker.sh
+        environment:
+            DOCKER_REPO: screwdrivercd/screwdriver
         secrets:
             # Publishing to NPM
             - NPM_TOKEN
             # Pushing tags to Git
-            - GIT_TOKEN
+            - GIT_KEY
```
Comment (d2lam) [inline screwdriver.yaml]: You created this yet? because I didn't.

Comment (d2lam) [inline screwdriver.yaml]: Just checked and saw that you created it.

File: screwdriver.yaml [hunk 2]
```diff
@@ -4,13 +4,17 @@ jobs:
         steps:
             - install: npm install
             - test: npm test
-            - coverage: ./ci/coverage
-            - publish: ./ci/publish
+            - setup-ci: git clone https://gist.github.com/3d2388b2a7ba658cdcdaffa8cd874e50.git ci
+            - coverage: ./ci/coverage.sh
+            - publish-npm: ./ci/publish.sh
```
Comment (d2lam) [inline screwdriver.yaml]: Should we make publish a separate job?

Comment (stjohnjohnson) [inline screwdriver.yaml]: Oooh, that's a GREAT idea

Comment (stjohnjohnson) [inline screwdriver.yaml]: Done.

**Pull request merged**